enhancements/ingress/set_eip_nlb_ingress.md

```diff
@@ -0,0 +1,894 @@
+    have minimum availability.), LoadBalancerReady=False (SyncLoadBalancerFailed:
+    The service-controller component is reporting SyncLoadBalancerFailed events like:
+    Error syncing load balancer: failed to ensure load balancer: error creating load 
+    balancer: Must have same number of EIP AllocationIDs (4) and SubnetIDs (5)
+   The kube-controller-manager logs may contain more details.)
+  reason: IngressControllerUnavailable
+  status: "False"
+  type: Available
+ - lastTransitionTime: "2024-04-20T06:40:43Z"
+  message: |-
+   One or more status conditions indicate progressing: DeploymentRollingOut=True
+    (DeploymentRollingOut: Waiting for router deployment rollout to finish: 0 
+    of 2 updated replica(s) are available...
+   )
+  reason: IngressControllerProgressing
+  status: "True"
+  type: Progressing
+ - lastTransitionTime: "2024-04-20T06:40:43Z"
+```
+
+- Check the status of `IngressController` controller for LoadBalancer which is LoadBalancerReady, LoadBalancerProgressing, Available.
+
+```sh
+  oc get ingresscontroller <ingresscontroller name> -o yaml
+ ```
+
+
+## Alternatives
+
+### Immutability
+
+Another alternative is to make the `eipAllocations` field immutable. This would require a
+cluster admin to delete and recreate the IngressController if they wanted to update
+the `eipAllocations`.
+
+The advantage of this approach is that the API Server would prevent any updates
+to the `eipAllocations` fields, clearly indicating to the cluster admin that they would
+need to recreate the IngressController. This is different from our current design,
+as described in [Effectuating EIPAllocation Updates](#effectuating-eip-updates-using-manual-method),
+which is more subtle. With the current design, a cluster admin would need to check the
+IngressController's status after updating `eipAllocations` for instructions on how to proceed.
+
+However, we didn't use the immutability design because of the following drawbacks:
+
+1. **Not Just a Load Balancer**: The IngressController, which encompasses DNS, HaProxy routers, and more, extends
+   beyond just a load balancer. Updating the eipAllocations on the Ingress Controller only needs to propagate to the
+   LB-type service, not the other components. Deleting the entire IngressController just to change `eipAllocations` is excessive,
+   considering that only the service itself requires deletion.
+2. **Prior Art**: We have an established a precedent in the [Ingress Mutable Publishing Scope](/enhancements/ingress/mutable-publishing-scope.md)
+   design, where cluster admins are allowed to mutate an IngressController API field, but must delete the Service to
+   effectuate the changes.
+3. **Future Possibilities**: Should the AWS CCM eventually supports reliably modifying subnets without deleting the
+   LB-type service, we can then easily adapt the Ingress Operator to immediately effectuate the `eipAllocations` and eliminate
+   the need for a cluster admin to delete the LB-type service.
```

Comment:
This is not a drawback, and isn't related to immutability.   It is really its own Alternative.
```suggestion
### Future Possibility based on changes to AWS CCM.

Should the AWS CCM eventually supports reliably modifying subnets without deleting the
   LB-type service, we can then easily adapt the Ingress Operator to immediately effectuate the `eipAllocations` and eliminate the need for a cluster admin to delete the LB-type service.
```